lightning/src/ln/channelmanager.rs

```diff
@@ -5611,22 +5676,25 @@ where
 
 				let monitor_res = self.chain_monitor.watch_channel(monitor.get_funding_txo().0, monitor);
 
-				let chan = e.insert(chan);
-				let mut res = handle_new_monitor_update!(self, monitor_res, peer_state_lock, peer_state,
-					per_peer_state, chan, MANUALLY_REMOVING_INITIAL_MONITOR,
-					{ peer_state.channel_by_id.remove(&new_channel_id) });
-
-				// Note that we reply with the new channel_id in error messages if we gave up on the
-				// channel, not the temporary_channel_id. This is compatible with ourselves, but the
-				// spec is somewhat ambiguous here. Not a huge deal since we'll send error messages for
-				// any messages referencing a previously-closed channel anyway.
-				// We do not propagate the monitor update to the user as it would be for a monitor
-				// that we didn't manage to store (and that we don't care about - we don't respond
-				// with the funding_signed so the channel can never go on chain).
-				if let Err(MsgHandleErrInternal { shutdown_finish: Some((res, _)), .. }) = &mut res {
-					res.0 = None;
+				if let ChannelPhase::Funded(chan) = e.insert(ChannelPhase::Funded(chan)) {
+					let mut res = handle_new_monitor_update!(self, monitor_res, peer_state_lock, peer_state,
+						per_peer_state, chan, MANUALLY_REMOVING_INITIAL_MONITOR,
+						{ peer_state.channel_by_id.remove(&new_channel_id) });
```

Comment:
I think it should still be fine here since extracting `chan` shouldn't make much of a difference to that line?

Reply:
Oh wait I see what you mean, like `chan` may be mutably borrowed twice because of the remove - though I guess we don't try to use `chan` after the remove because locally extracting it works fine